### PR TITLE
Fix/exam backend

### PR DESCRIPTION
**Pull request by NansD**
This branch just applies some fixes on the backend.
I wanted to use the mongoose $pull technique, but it doesn't remove the subdocument from the database so this operation would leave use with plenty of "exercices" without any parent exam and so on.
(https://docs.mongodb.com/manual/reference/operator/update/pull/ && https://mongoosejs.com/docs/api.html#mongoosearray_MongooseArray-pull )

Note that now with your restclient, you have to use JSON to test the backend : 
image
image

### CODE REVIEW

File: backend/src/api/ressources/exams/controller.js [hunk 1]
```diff
@@ -44,27 +44,25 @@ const controller = {
...
+    const { exam } = res.locals;
     if (req.body) {
       try {
-        const result = await ExamData.update(req.params.examId, req.body);
-        if (!result) {
-          return res.status(404).send('Exam not Found');
-        }
-        return res.status(200).json(result);
+        exam.update(req.body);
+        await exam.save();
```
Comment (bdebever) [inline backend/src/api/ressources/exams/controller.js:64]: AFAIK, exam.update() returns a promise. https://mongoosejs.com/docs/documents.html

So we should `await`

**Review by bdebever — COMMENTED** [review]
I have added one comment on the update. 
Otherwise, all good to me!

**Review by ValentinChion — APPROVED** [review]
Great work here as always :rocket:

File: backend/src/api/ressources/exams/controller.js (hunk 1 above)
Comment (NansD) [reply backend/src/api/ressources/exams/controller.js:64]: as far as I tested, it works properly
document.update in mongoose is not asynchronous, as it does not perform any operation in the database. It only updates the document variable and checks for model validation.
The operation in the database is done with `exam.save()`
More info : https://mongoosejs.com/docs/api.html#document_Document-update

**Review by nymous — APPROVED** [review]

**Pull request merged**